### ISSUE TITLE
Add support for state files

### ISSUE THREAD

**Issue by tbohn** [UW-Hydro/tonic#35]
Tonic needs to be able to convert state files, for users to be able to switch from older versions to 5.0 or from classic to image mode.

In addition, it would be helpful to add the capability of generating an initial state file.  This could allow us to remove that function from VIC itself.  Once tonic has the ability to convert state files between ascii and netcdf, we would just need to add a function to define state variables, which would then be written to netcdf (or ascii, I suppose).  The other parameter files would need to be read in for tonic to know the set of states that exist in each grid cell.  For the default values, we could either port the logic currently in VIC or we could allow the user to specify the (a) fraction of maximum storage to initialize to (constant across all moisture storages) and (b) the initial temperature (constant across all temperatures).

**Comment by jhamman** [comment]
:+1:

**Comment by tbohn** [comment]
The ascii-netcdf conversion aspect of this issue is being addressed in PR #43.  The netcdf-ascii conversion aspect of this issue has been moved into a new issue #44.  Thus, this issue can be closed once PR #43 is merged.
